eng/tools/generate-dev-builds.js

```diff
@@ -0,0 +1,241 @@
+const path = require("path");
+const jju = require("jju");
+const util = require("util");
+const fs = require("fs");
+const writeFile = util.promisify(fs.writeFile);
+const readFile = util.promisify(fs.readFile);
+
+const appendPackageData = (data, pkgSrc, pkgJson) => {
+
+  var dep = new Map();
+  for(var key in pkgJson.dependencies){
+    dep.set(key, pkgJson.dependencies[key]);
+  }
+
+  var devDep = new Map();
+  for(var key in pkgJson.devDependencies){
+    devDep.set(key, pkgJson.devDependencies[key]);
+  }
+
+  data[pkgJson.name] = {
+    src: pkgSrc,
+    ver: pkgJson.version,
+    dep: dep,
+    devDep: devDep,
+    visited: false,
+    visitedDep : false
+  };
+};
+
+const getRushPackages = async (rushPath) => {
+  const baseDir = path.dirname(rushPath);
+  const rushJson = jju.parse(await readFile(rushPath, "utf8"));
+  const packageData = {};
+
+  for (const proj of rushJson.projects) {
+    const projDir = path.join(baseDir, proj.projectFolder);
+    const packageJson = jju.parse(await readFile(path.join(projDir, "package.json"), "utf8"));
+    appendPackageData(packageData, projDir, packageJson);
+  }
+
+  return packageData;
+};
+
+function writeToPackageJson(pckJsonPath,contents){
+  console.log("Inside writeTo packageJson function for - " + pckJsonPath );
+  var data = JSON.stringify(contents,null,2);
+  try{
+      fs.access(pckJsonPath, fs.F_OK, (err) => {
+      if (err) {
+        console.error(err)
+        return
+      }
+    });
+
+    fs.writeFileSync(pckJsonPath, data);
+    console.info("File created successfully with Node.js v10 fs/promises!");
+  }
+  catch(e){
+    console.error(`\n\n${e.toString()}\n\n`);
+    process.exit(1);
+  }
+  
+}
+
+function editVersion(rushPackages, buildNo,package){
+  console.log("Inside editVersion function - "+ package);
+  var pckJsonPath = rushPackages[package].src+"/"+"package.json";
+  console.log("attempting to read..");
+  console.log(pckJsonPath);
+  var contents = JSON.parse(fs.readFileSync(pckJsonPath).toString());
+  contents.version = contents.version + "-dev-"+buildNo;
+  console.log(contents);
+  writeToPackageJson(pckJsonPath,contents);
+  rushPackages[package].ver = contents.version;
+  //console.log(rushPackages[package]);
+  return rushPackages;
+}
+
+function editDependencyVersion(rushPackages,package,dep,version,depType){
```

Comment:
Last I chatted with @mitchdenny I think the idea here is that we wouldn't need to update the dependencies based on our dev versioning. The only ones that we would ever even need to consider updating would be ones that aren't published anywhere and once we start publishing nightly builds people will need to start depending on the nightly builds if they want the latest changes and not depending on the version we build in the repo.

Reply:
removed the dependency version updates from the code